app/controllers/discover.js

```diff
@@ -121,6 +127,39 @@ export default ApplicationController.extend({
         });
     },
 
+    transformTypes(obj) {
+        if (typeof (obj) !== 'object') {
+            return obj;
+        }
+        var keys = Object.keys(obj);
+        var n = keys.length;
+
+        var lowKey;
+        while (n--) {
+            var key = keys[n];
+            if (key === (lowKey = key.replace(/([A-Z])/g, ' $1').trim().toLowerCase())) {
+                obj[key] = this.transformTypes(obj[key]);
+                continue;
+            }
+            obj[lowKey] = this.transformTypes(obj[key]);
+            delete obj[key];
+        }
+        return (obj);
+    },
+
+    getTypes() {
```

Comment:
Could you add an if guard here? If someone revisits this page without actually refreshing this request does not need to be made again.